libraries/mne/c/mne_msh_display_surface_set.cpp

```diff
@@ -442,21 +442,14 @@ int MneMshDisplaySurfaceSet::add_bem_surface(MneMshDisplaySurfaceSet* surfs,
     newSurf->overlay_color_mode   = SHOW_OVERLAY_HEAT;
 
     decide_surface_extent(newSurf,name);
-    add_replace_display_surface(surfs,newSurf,TRUE,TRUE);
+    add_replace_display_surface(surfs, newSurf, true, true);
     apply_left_eyes(surfs);
     setup_current_surface_lights(surfs);
 
     return OK;
 
 bad : {
-        if(surf)
-        {
-            delete surf;
-        }
-        if(newSurf)
-        {
-            delete newSurf;
-        }
```

Comment:
Why are we not cleaning up newSurf anymore?

Reply:
Yes. We're on that. This  is WP. 
But before getting on with it, I'd really need some help trying to understand what is going on with ```add_repace_display_surface``` which also (probably) has a leak. I'd like to solve both issues at the same time. 
Could you help me understand what is this fcn doing?

here: https://github.com/mne-tools/mne-cpp/blob/edffebb123ce116e5bd1ef027acb8a9a217cb8c0/libraries/mne/c/mne_msh_display_surface_set.cpp#L459

Reply:
I guess we parse through a set of display surfaces and if we find a match (based on the ID) we delete the old surface and replace it with a new one?

Reply:
Thank you.